### PR TITLE
add crud for general function user process, add pinFriend

### PR DESCRIPTION
**Pull request by AndrewZuo01**
<br>

#### 👀 What this PR does / why we need it:

- [x ] My pull request adheres to the code style of this project
- [x ] My code requires changes to the documentation
- [x ] I have updated the documentation as required
- [x ] All the tests have passed

add crud for general function user process. For example, add to favorite

#### 🅰 Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: internal/rpc/friend/friend.go [hunk 1]
```diff
@@ -434,3 +435,36 @@ func (s *friendServer) GetSpecifiedFriendsInfo(ctx context.Context, req *pbfrien
...
+	if utils.Duplicate(req.FriendUserIDs) {
+		return nil, errs.ErrArgs.Wrap("friendIDList repeated")
+	}
+	var isPinned bool
+	if req.IsPinned != nil {
+		isPinned = req.IsPinned.Value
+	} else {
+		return nil, errs.ErrArgs.Wrap("isPinned is nil")
+	}
+	//檢查是不是在好友列表裏
```
Comment (cubxxw) [inline internal/rpc/friend/friend.go:454]: Why use Chinese comments?

File: internal/api/route.go [hunk 2]
```diff
@@ -77,6 +77,11 @@ func NewGinRouter(discov discoveryregistry.SvcDiscoveryRegistry, rdb redis.Unive
 		userRouterGroup.POST("/subscribe_users_status", ParseToken, u.SubscriberStatus)
 		userRouterGroup.POST("/get_users_status", ParseToken, u.GetUserStatus)
 		userRouterGroup.POST("/get_subscribe_users_status", ParseToken, u.GetSubscribeUsersStatus)
+
+		userRouterGroup.POST("/process_user_command_add", ParseToken, u.ProcessUserCommandAdd)
+		userRouterGroup.POST("/process_user_command_delete", ParseToken, u.ProcessUserCommandDelete)
```
Comment (cubxxw) [inline internal/api/route.go:82]: + [ ] add api code 
+ [ ] add api docs
+ [ ] add api aotu test (cicd)

**Review by cubxxw — CHANGES_REQUESTED** [review]

File: internal/rpc/friend/friend.go (hunk 1 above)
Comment (AndrewZuo01) [reply internal/rpc/friend/friend.go:454]: because other comments are also Chinese

**Review by cubxxw — APPROVED** [review]

**Review by cubxxw — CHANGES_REQUESTED** [review]
CICD error

**Review by cubxxw — APPROVED** [review]

**Review by AndrewZuo01 — COMMENTED** [review]
please merge pin friend

**Review by cubxxw — APPROVED** [review]

**Review by FGadvancer — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -154,3 +154,7 @@ require (
 	golang.org/x/crypto v0.14.0 // indirect
 	gopkg.in/ini.v1 v1.67.0 // indirect
 )
+//replace (
+//	github.com/OpenIMSDK/protocol v0.0.34 => github.com/AndrewZuo01/protocol v0.0.0-20231218034338-b8d838e0b182
```
Comment (cubxxw) [inline go.mod:158]: ??
why?

**Review by cubxxw — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (AndrewZuo01) [reply go.mod:158]: for easy to modify

**Pull request merged**